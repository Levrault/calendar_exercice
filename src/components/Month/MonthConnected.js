import { connect } from 'react-redux';
import Month from './Month';
import { eventsByMonthIdFilter } from '../Event/Events-reselect';

/**
 * @param {object} state
 * @param {object} ownProps
 * @returns {object}
 */
const mapStateToProps = (state, ownProps) => {
  return {
    events: eventsByMonthIdFilter(state, ownProps)
  };
};

export default connect(mapStateToProps)(Month);
