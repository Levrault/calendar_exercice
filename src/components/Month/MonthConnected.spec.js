import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import getFakeStore from '../../tools/test/fakeStore';
import getFakeData from '../../tools/test/fakeData';
import MonthConnected from './MonthConnected';

const store = getFakeStore();
const mock = getFakeData().year.data[0].months[1];

describe('MonthConnected component', () => {
  it('should render without problem', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MonthConnected {...mock} />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
