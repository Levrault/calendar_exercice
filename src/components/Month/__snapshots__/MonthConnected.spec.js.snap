// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthConnected component should render without problem 1`] = `
<div>
  <h1>
    February
  </h1>
  <div>
    <ul>
      <li>
        S
      </li>
      <li>
        M
      </li>
      <li>
        T
      </li>
      <li>
        W
      </li>
      <li>
        T
      </li>
      <li>
        F
      </li>
      <li>
        S
      </li>
    </ul>
    <ul>
      <li />
      <li />
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            1
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            2
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            3
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            4
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            5
          </button>
        </div>
      </li>
    </ul>
    <ul>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            6
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            7
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className="undefined"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#EEEEEE",
                "color": "#212121",
              }
            }
          >
            8
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            9
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            10
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            11
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            12
          </button>
        </div>
      </li>
    </ul>
    <ul>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            13
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            14
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            15
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            16
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            17
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            18
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className="undefined"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#546E7A",
                "color": "#212121",
              }
            }
          >
            19
          </button>
        </div>
      </li>
    </ul>
    <ul>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            20
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            21
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            22
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            23
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            24
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            25
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            26
          </button>
        </div>
      </li>
    </ul>
    <ul>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            27
          </button>
        </div>
      </li>
      <li>
        <div>
          <button
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            28
          </button>
        </div>
      </li>
      <li />
      <li />
      <li />
      <li />
      <li />
    </ul>
  </div>
</div>
`;
